plugins/modules/edpm_derive_pci_passthrough_devicespec.py

```diff
@@ -0,0 +1,650 @@
+        phy = phys_map.strip()
+        map = re.split(',', phy)
+        for i in map:
+            net_name, nic_name = i.split(':')
+            if net_name == physical_network:
+                return nic_name
+    return None
+
+
+def generate_combined_configuration(user_configs, system_configs,
+                                    phys_dev_map=None):
+    """Derived configuration = user_config - system_configs
+
+    Identify the user_defined configuration that overlaps with the
+    NIC Partitioned VFs and remove those VFs from the derived configuration
+    In case of no overlap, the user defined configuration shall be used
+    as it is.
+    :param user_configs: device_spec list from nova conf
+    :param system_configs: Derived from sriov-mapping.yaml
+    :param phys_dev_map: Sriov physical device mannping
+    :return user_config_copy: Any non-nic partinioned cfg will be returned in this list
+    :return derived_config: All nic partinioned cfg will be returned after derivation in this list
+    """
+
+    user_config_copy = []
+    derived_config = []
+
+    allocated_pci = get_allocated_pci_addresses(system_configs)
+    nic_partition_pfs = get_sriov_nic_partition_pfs(system_configs)
+
+    for user_config in user_configs:
+        if 'address' in user_config and 'devname' in user_config:
+            msg = f"Both devname and address can't be present in {_PASSTHROUGH_DEVSPEC_KEY}"
+            raise InvalidConfigException(msg)
+
+        keys = ['address', 'product_id', 'devname']
+        if not any(k in user_config for k in keys):
+            # address or product_id or devname not present in user_config
+            pf = get_pf_name_from_phy_network(user_config['physical_network'], phys_dev_map)
+            user_config['address'] = get_pci_addr_from_ifname(pf)
+
+        if 'devname' in user_config:
+            if user_config['devname'] in nic_partition_pfs:
+                user_config['address'] = get_pci_addr_from_ifname(user_config['devname'])
+                del user_config['devname']
+            else:
+                user_config_copy.append(user_config)
+                continue
+        if ('address' in user_config
+                or 'product_id' in user_config):
+            passthrough_tmp, del_user_config = get_passthrough_config_for_all_pf(
+                user_config, system_configs, allocated_pci)
+
+            """ If del_user_config is set, do not add to derived_config or
+                user_config_copy
+            """
+            if not del_user_config:
+                if passthrough_tmp:
+                    derived_config.extend(passthrough_tmp)
+                else:
+                    user_config_copy.append(user_config)
+        else:
+            user_config_copy.append(user_config)
+    return (user_config_copy, derived_config)
+
+
+def get_nova_userconfigs(nova_conf_file):
+    configs = []
+    src_config = ""
+    # Currently only "device_spec" is valid
+    valid_spec = ('device_spec')
```

Comment:
@vcandapp  can we reuse this variable PASSTHROUGH_DEVSPEC_KEY instead of valid_spec new variable again?